source/schwarz_base.cpp

```diff
@@ -391,6 +438,38 @@ void SchwarzBase<ValueType, IndexType>::run(
                 4, metadata.my_rank, expand_local_vec, metadata.iter_count);
         }
     }
+
+    //CHANGED
+    //Closing file
+    fps.close();
```

Comment:
All the file operations should be out of the timed loop. You can write them into variables, but file I/O will terribly affect the timings